metricbeat/module/nginx/stubstatus/data.go

```diff
@@ -0,0 +1,91 @@
+package stubstatus
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Map body to MapStr
+func eventMapping(m *MetricSet, body io.ReadCloser, hostname string, metricset string) (common.MapStr, error) {
+	// Nginx stub status sample:
+	// Active connections: 1
+	// server accepts handled requests
+	//  7 7 19
+	// Reading: 0 Writing: 1 Waiting: 0
+	var (
+		active   int
+		accepts  int
+		handled  int
+		dropped  int
+		requests int
+		current  int
+		reading  int
+		writing  int
+		waiting  int
+	)
+
+	var re *regexp.Regexp
+	scanner := bufio.NewScanner(body)
+
+	// Parse active connections.
+	scanner.Scan()
+	re = regexp.MustCompile("Active connections: (\\d+)")
```

Comment:
I recommend declaring all the regular expressions as package level variables. Currently this is rebuilding the regular expression each time it parses. It also defers the validation of the regular expression until the method is invoked.

So if you move all the `regexp.MustCompile` calls into a `var (   )` block, then the regular expressions are only compiled once (performance boost) and they are validated at start-up irregardless of whether the module is used.